examples/authentication/proto/authentication.proto

```diff
@@ -0,0 +1,32 @@
+//
+// Copyright 2020 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+syntax = "proto3";
+
+package oak.examples.authentication;
+
+import "google/protobuf/empty.proto";
+
+// Parameters needed by clients to initiate OpenID Connect authentication requests.
+message AuthParameters {
+  string client_id = 1;
+  string auth_endpoint = 2;
+  // TODO(#886): Add support for scope, state and code challenge.
+}
+
+service Authentication {
+  rpc GetAuthParameters(google.protobuf.Empty) returns (AuthParameters);
+}
```

Comment:
I don't immediately see why we need a gRPC server for this, as opposed to just an endpoint that returns a single string which is the URL to open in the browser? Is there value in preserving all this structure?

Reply:
The `redirect_uri` querystring parameter in the resulting URL is dependent on the client, which is why I decided to let the client construct the final URL. It could be handled by passing the `redirect_uri` as a paramter to the auth server, or by including a token that the client replaces. I don't think either of these would make the code much clearer though.

Also, the next step is to exchange the code for a token, and validating the token. This can once again be done with URL parameters and a get request that returns the token, but I don't think manually handling URL parameters and base64 (or JSON) encoding and decoding of the token bytes will make the code clearer or simpler.